docs/release-notes-ps-v8.0.31.md

```diff
@@ -0,0 +1,70 @@
+
+    Setting `sql_generate_invisible_primary_key` on the source is ignored by a replica because this variable is not replicated. This behavior is inherited from the previous releases.
+
+* MySQL adds support for the SQL standard [`INTERSECT`](https://dev.mysql.com/doc/refman/8.0/en/intersect.html) and [`EXCEPT`](https://dev.mysql.com/doc/refman/8.0/en/except.html) table operators.
+
+* InnoDB supports parallel index builds. This improves index build performance. The sorted index entries are loaded into a B-tree in a multithread. In previous releases, this action was performed by a single thread.
+
+* The Performance and sys schemas show metrics for the global and session memory limits introduced in MySQL 8.0.28.
+
+    The following columns have been added to the Performance Schema tables:
+
+    | Performance Schema tables                                                            | Columns                                                   |
+    | ------------------------------------------------------------------------------------ | --------------------------------------------------------- |
+    | SETUP_INSTRUMENTS                                                                    | FLAGS                                                     |
+    | THREADS                                                                              | CONTROLLED_MEMORY, MAX_CONTROLLED_MEMORY, TOTAL_MEMORY, MAX_TOTAL_MEMORY |
+    | EVENTS_STATEMENTS_CURRENT, EVENTS_STATEMENTS_HISTORY, EVENTS_STATEMENTS_HISTORY_LONG | MAX_CONTROLLED_MEMORY, MAX_TOTAL_MEMORY                   |
+    | Statement Summary Tables                                                             | MAX_CONTROLLED_MEMORY, MAX_TOTAL_MEMORY                   |
+    | Performance Schema Connection Tables                                                 | MAX_SESSION_CONTROLLED_MEMORY, MAX_SESSION_TOTAL_MEMORY   |
+    | PREPARED_STATEMENTS_INSTANCES                                                        | MAX_CONTROLLED_MEMORY, MAX_TOTAL_MEMORY                   |
+
+    The following columns have been added to the sys schema `STATEMENT_ANALYSIS` and `X$STATEMENT_ANALYSIS` views:
+
+    * MAX_CONTROLLED_MEMORY
+
+    * MAX_TOTAL_MEMORY
+
+    The `controlled_by_default` flag has been added to the `PROPERTIES` column of the `SETUP_INSTRUMENTS` table.
+
+    Now, you can add and remove non-global memory instruments to the set of controlled-memory instruments. To do this, set the value of the `FLAGS` column of `SETUP_INSTRUMENTS`.
+
+    ```sql
+    SQL> UPDATE PERFORMANCE_SCHEMA.SETUP_INTRUMENTS SET FLAGS="controlled" 
+    WHERE NAME='memory/sql/NET::buff';
+    ```
+
+* The `audit_log_flush` variable has been deprecated and will be removed in future releases.
+
+Find the full list of bug fixes and changes in the [MySQL 8.0.31 Release Notes](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-31.html).
+
+Review each component’s release notes for What’s new, improvements, or bug fixes.
```

Comment:
Make this the first paragraph of the next header. 
Otherwise. Good.

Reply:
Uodated